### PR TITLE
fix unskipped fields at the root of the selection set #930

### LINKED ISSUE #930: Bug on repeated fields with different skip and include directive value
Here is a difference detected by apollo-smith
```
====DOCUMENT===
query {
  allPandas @skip(if: false) @include(if: false) {
    name
    favoriteFood
  }
  allPandas @skip(if: false) {
    name
    name
  }
}


====GATEWAY====
{
  "data": {
    "allPandas": [
      {
        "name": "Basi"
      },
      {
        "name": "Yun"
      }
    ]
  }
}

====ROUTER====
{
  "data": {
    "allPandas": [
      {
        "favoriteFood": null,
        "name": "Basi"
      },
      {
        "favoriteFood": null,
        "name": "Yun"
      }
    ]
  }
}
```

### PR DESCRIPTION
**Pull request by bnjjj**
close #930 

I also added support of fed2 for fuzz targets, because we used fed1 before, it's better to target fed2.

### CODE REVIEW

File: apollo-router-core/src/spec/query.rs [hunk 1]
```diff
@@ -479,9 +479,27 @@ impl Query {
...
-                    skip: _,
-                    include: _,
+                    skip,
+                    include,
                 } => {
+                    if skip
+                        .should_skip(variables)
+                        // validate_variables should have already checked that
+                        // the variable is present and it is of the correct type
+                        .unwrap_or(false)
```
Comment (o0Ignition0o) [inline apollo-router-core/src/spec/query.rs:489]:
```suggestion
                        .unwrap_or_default()
```

File: docker-compose-federation2.yml [hunk 2]
```diff
@@ -1,6 +1,19 @@
...
+  gateway:
+    container_name: apollo-gateway
+    build: dockerfiles/federation2-demo/gateway
+    environment:
+      - APOLLO_OTEL_EXPORTER_TYPE=collector
+      - APOLLO_OTEL_EXPORTER_HOST=collector
+      - APOLLO_OTEL_EXPORTER_PORT=4318
+      - APOLLO_SCHEMA_CONFIG_EMBEDDED=true
+    volumes:
+      - ./dockerfiles/federation2-demo/supergraph.graphql:/etc/config/supergraph.graphql
```
Comment (o0Ignition0o) [inline docker-compose-federation2.yml:13]: is this intentional? not sure how this ties to the pull request

**Review by o0Ignition0o — APPROVED** [review]
Great job!

File: NEXT_CHANGELOG.md [hunk 3]
```diff
@@ -66,6 +66,9 @@ telemetry:
 ```
 
 ## 🐛 Fixes
+### Fields in the root selection set of a query is now correctly skipped and included [PR #931](https://github.com/apollographql/router/pull/931)
```
Comment (garypen) [inline NEXT_CHANGELOG.md:69]:
```suggestion
### Fields in the root selection set of a query are now correctly skipped and included [PR #931](https://github.com/apollographql/router/pull/931)
```

File: docker-compose-federation2.yml (hunk 2 above)
Comment (garypen) [reply docker-compose-federation2.yml:13]: Yep. This looks like it was included accidentally.

**Review by garypen — CHANGES_REQUESTED** [review]

File: docker-compose-federation2.yml (hunk 2 above)
Comment (bnjjj) [reply docker-compose-federation2.yml:13]: No it's intentional, I need it when running apollo-smith, in the fed1 version that was done like that

**Review by Geal — APPROVED** [review]

**Pull request merged**